Docker: Add the Self Service UI

The SSUI is missing from the Docker image.  The manageiq-ui-self_service repo would need to be cloned and properly built as we do for the VM builds.

cc @bazulay @fbladilo

I'm having some issues when npm installing at /opt/manageiq/manageiq-ui-self_service , it complains that the installed karma module is too new for gulp-karma and fails : 

[root@3ddc47ca870c manageiq-ui-self_service]# npm install
npm WARN package.json gulp-minify-css@1.2.4 No repository field.
npm WARN package.json gulp-minify-html@1.0.6 No repository field.
npm WARN cannot run in wd manageiq-ui-self_service@0.0.1 bower install -F (wd=/opt/manageiq/manageiq-ui-self_service)
npm WARN cannot run in wd manageiq-ui-self_service@0.0.1 gulp build (wd=/opt/manageiq/manageiq-ui-self_service)
npm ERR! peerinvalid The package karma does not satisfy its siblings' peerDependencies requirements!
npm ERR! peerinvalid Peer gulp-karma@0.0.4 wants karma@>=0.10 <=0.13
npm ERR! peerinvalid Peer karma-chai@0.1.0 wants karma@>=0.10.9
npm ERR! peerinvalid Peer karma-chai-sinon@0.1.5 wants karma@>=0.6
npm ERR! peerinvalid Peer karma-coverage@0.2.7 wants karma@>=0.9
npm ERR! peerinvalid Peer karma-growl-reporter@0.1.1 wants karma@>=0.9
npm ERR! peerinvalid Peer karma-mocha@0.1.10 wants karma@>=0.12.8
npm ERR! peerinvalid Peer karma-phantomjs-launcher@0.1.4 wants karma@>=0.9
npm ERR! peerinvalid Peer karma-sinon@1.0.4 wants karma@>=0.10

Karma installed is : 

[root@3ddc47ca870c manageiq-ui-self_service]# npm view karma | grep version:
  version: '0.13.22',

Did I miss something? @Fryguy @bdunne @bazulay

--- a/Dockerfile
+++ b/Dockerfile
@@ -7,6 +7,7 @@ ENV LANG en_US.UTF-8
 ENV TERM xterm
 ENV APP_ROOT /var/www/miq/vmdb
 ENV APPLIANCE_ROOT /opt/manageiq/manageiq-appliance
+ENV SSUI_ROOT /opt/manageiq/manageiq-ui-self_service
 
 # Fetch postgresql 9.4 COPR and pglogical repos
 RUN curl -sSLko /etc/yum.repos.d/rhscl-rh-postgresql94-epel-7.repo \
@@ -71,8 +72,10 @@ RUN curl -sL https://github.com/postmodern/chruby/archive/v0.3.9.tar.gz | tar xz
     echo "chruby ruby-2.2.4" >> ~/.bash_profile && \
     rm -rf /chruby-* && rm -rf /usr/local/src/* && yum clean all
 
-## GIT clone manageiq-appliance repo
-RUN git clone --depth 1 https://github.com/ManageIQ/manageiq-appliance.git ${APPLIANCE_ROOT}
+## GIT clone manageiq-appliance and self-service UI repo (SSUI)
+RUN git clone --depth 1 https://github.com/ManageIQ/manageiq-appliance.git ${APPLIANCE_ROOT} && \
+git clone --depth 1 https://github.com/ManageIQ/manageiq-ui-self_service.git ${SSUI_ROOT} && \
+ln -vs ${APP_ROOT} /opt/manageiq/manageiq
 
 ## Create approot, ADD miq
 RUN mkdir -p ${APP_ROOT}
@@ -91,12 +94,22 @@ echo "export APP_ROOT=${APP_ROOT}" >> /etc/default/evm
 WORKDIR ${APP_ROOT}
 RUN source /etc/default/evm && \
 /usr/bin/memcached -u memcached -p 11211 -m 64 -c 1024 -l 127.0.0.1 -d && \
-npm install -g bower && \
+npm install npm -g && \
+npm install gulp bower -g && \
 gem install bundler -v ">=1.8.4" && \
 bin/setup --no-db --no-tests && \
 rake evm:compile_assets && \
 rake evm:compile_sti_loader && \
 rm -rvf /opt/rubies/ruby-2.2.4/lib/ruby/gems/2.2.0/cache/* && \
+bower cache clean && \
+npm cache clean
+
+## Build SSUI
+RUN source /etc/default/evm && \
+cd ${SSUI_ROOT} && \
+npm install && \
+bower -F --allow-root install && \
+gulp build && \
 bower cache clean && \
 npm cache clean
 

